### PR TITLE
adding option to create a route to nat gateway in database subnets

### PR DESCRIPTION
**Pull request by ebarault**
Similarly to `create_database_internet_gateway_route`, this adds the `create_database_nat_gateway_route` parameter to create routes in database subnets to the unique/per-AZ/per-subnet NAT Gateways.

### CODE REVIEW

**Comment by ebarault** [comment]
@antonbabenko any feedback?

File: main.tf [hunk 1]
```diff
@@ -133,6 +133,18 @@ resource "aws_route" "database_internet_gateway" {
   }
 }
 
+resource "aws_route" "database_nat_gateway" {
+  count = "${var.create_vpc && var.enable_nat_gateway && var.create_database_subnet_route_table && length(var.database_subnets) && var.create_database_nat_gateway_route && !var.create_database_internet_gateway_route > 0 ? local.nat_gateway_count : 0}"
```
Comment (antonbabenko) [inline main.tf]: There is an error in expression. Should be `"${var.create_vpc && var.enable_nat_gateway && var.create_database_subnet_route_table && length(var.database_subnets) > 0 && var.create_database_nat_gateway_route && !var.create_database_internet_gateway_route ? local.nat_gateway_count : 0}"`.

The rest looks good.

**Comment by ebarault** [comment]
@antonbabenko arff you're right, sorry about that, i just pushed the fix, please have a look

**Comment by antonbabenko** [comment]
Thank you, Eric!

[v1.53.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v1.53.0) has been just released.

**Pull request merged**